app/components/Login.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import { useState } from 'react';
+import styled from 'styled-components';
+import { Redirect } from 'react-router-dom';
+const ipcRenderer = require('electron').ipcRenderer;
+
+const LoginContainer = styled.div`
+  background-color: #e8ecf1;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  padding: 50px;
+  height: 100vh;
+`
+const URIInput = styled.textarea`
+  width: 200px;
+  height: 100px;
+  border-radius: 3px;
+  border: ${props => props.requiredErr ? '1px solid #ca333e' : '1px solid lightgrey'};
+  overflow: wrap;
+  resize: none;
+  transition: 0.3s;
+  padding: 5px;
+  font-family: 'Poppins', sans-serif;
+  letter-spacing: 2px;
+
+  :focus {
+    outline: none;
+  }
+`
+const LoginBtn = styled.button`
+  padding: 5px 20px;
+  margin: 10px;
+  font-family: 'Poppins', sans-serif;
+  display: flex;
+`
+
+const InputLabel = styled.span`
+  color: black;
+  font-family: 'Poppins', sans-serif;
+`
+
+const ConnectionErrorMessage = styled.div`
+  background-color: #f1c7ca;
+  color: #ca333e;
+  border-radius: 3px;
+  padding: 10px;
+  margin: 5px;
+  font-size: 80%;
+  font-family: 'Poppins', sans-serif;
+  transition: 0.5s;
+`
+
+const RequiredWarning = styled.span`
+  color: #ca333e;
+  font-size: 60%;
+`
+
+//Functional component 
```

Comment:
"only commit comments explaining the 'why' not the 'what'" a sage soul once said 

+ we're (going to be?) adding types where `React.SFC`'ll do